internal/websockets/gamesocket.go

```diff
@@ -56,35 +56,47 @@ func (g *GameSocket) Close() error {
 // ServeHTTP handles websocket requests for GameSocket - JoinGameEvent
 // and connects the user to a game session if correct id is produced
 func (g *GameSocket) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	idList, ok := r.URL.Query()[queryGameID]
+	ctx := r.Context()
+	player := ctx.Value(api.DecodedUserKey).(*game.Player)
+
+	idList, ok := r.URL.Query()[queryRoomID]
 	if !ok || len(idList) < 1 {
-		fmt.Println("error getting game id")
+		fmt.Println("error getting room id")
 		return
 	}
 
-	id := idList[0]              // get the first id
-	_uuid, err := uuid.Parse(id) // convert to UUID
-	if err != nil {
-		fmt.Println("error parsing game id")
+	id := idList[0] // get the first id
+	room := g.rooms[id]
+	canJoinGame := true
+	// check if the room exists
+	if room == nil {
+		api.BadRequest(fmt.Errorf("room with id: %v does not exist", id).Error()).WriteJSON(w)
 		return
 	}
+	// check if the room has an active game
+	if room.hasActiveGame {
+		canJoinGame = false
+		for key := range room.players {
+			if key.playerId == player.ID.String() {
+				canJoinGame = true
+			}
```

Comment:
tests need to written for this section because there is a lot going on here. Also, i think the middleware that attaches a user to the Context is only available for the api endpoints, so we have to bring it up, to use for the websockets as well.